### ISSUE TITLE
Would you like to support ripgrep?

### ISSUE THREAD

**Issue by yuzumx** [NicolasPetton/xref-js2#12]
Hello:

I would like to ask, are you willing to support ripgrep?

I now have to install both ag and rg..

Thank you.

**Comment by NicolasPetton** [comment]
MX Reimu <notifications@github.com> writes:

> I would like to ask, are you willing to support ripgrep?

Sure!

Do you want to do it and open a PR?

Cheers,
Nico

**Comment by yuzumx** [comment]
Well ... I'm a junk rookie……emacs-lisp like hieroglyphics to me.....

Sorry!

**Comment by mohkale** [comment]
seeing as ag and rg have the same output format, couldn't we just replace `xref-js2-ag-arguments` with `xref-js2-search-command` with the car of the list being the command and the cdr being the arguments. that way users can easily move to whatever grep program they want (so long as it conforms to the same output format). Anyways, I'll look into it and try to make a pull request.

**Comment by NicolasPetton** [comment]
Yes, this seems like a good idea.

**Comment by mohkale** [comment]
oh, just to clarify ag and rg have different command line arguments so that wouldn't have worked (because of `xref-js2-ignored-dirs` and `xref-js2-ignored-files`). I haven't used either very extensively outside of emacs so I wasn't aware. In the pull request I've made [17](https://github.com/NicolasPetton/xref-js2/pull/17) I just added a few new variables allowing users to customise the search operation. `xref-js2-search-program` is how you specify you want to use rg instead of ag and `xref-js2-rg-arguments` is how you specify the arguments passed to rg. By default it still uses ag and when using rg it should still give the same results as ag.
